File: CMOC.Domain/Equipment.cs
namespace CMOC.Domain;

public class Equipment
{
    public int Id { get; set; }
    public string SerialNumber { get; set; }
    public List<ComponentRelationship> Components { get; set; }
    public int TypeId { get; set; }
    public EquipmentType Type { get; set; }
    public int LocationId { get; set; }
    public Location Location { get; set; }
    public int? IssueId { get; set; }
    public Issue Issue { get; set; }
    public List<ServiceSupportRelationship> Relationships { get; set; }
    public string Notes { get; set; }
    public bool? OperationalOverride { get; set; }
}

File: CMOC.UI/src/types/Equipment.ts
import {Issue} from "./Issue.ts";
import {Component} from "./Component.ts";

export type Equipment = {
    id: number;
    serialNumber: string;
    components: Component[];
    typeId: number;
    typeName: string;
    locationId: number;
    location: string;
    supportedServices: number[];
    notes: string | null;
    operationalOverride: boolean | null;
    issueId: number | null;
    issue: Issue | null;
    status: number;
}